plugin/makeshift.vim

```diff
@@ -87,6 +88,12 @@ function! s:makeshift()
     if len(l:program) == 0
         let l:program = s:determine_build_system(expand('%:p:h'))
     endif
+    if l:program == 'waf'
+        let l:found = globpath(b:makeshift_root, l:program)
+        if filereadable(l:found)
+            let l:program = l:found
+        endif
+    endif
```

Comment:
I have some questions around this:

- Should finding a bundled `waf` be a concern of Makeshift or should it be handled by the OS path?
- If finding a bundled `program` _is_ a concern of this plugin, should this functionality to be specific to `waf` or could we remove that condition?
- If finding a bundled `program` _is_ a concern of this plugin, does it deserve its own function?
- If finding a bundled `program` _is_ a concern of this plugin, should there be a configuration option to disable it?